build-script.py

```diff
@@ -661,115 +768,117 @@ def run_xctests(toolchain, build_dir, multiroot_data_file, release, verbose):
+            "directory.",
+        )
+
+        parser.add_argument(
+            "--disable-sandbox",
+            action="store_true",
+            help="Disable sandboxes when building with SwiftPM",
+        )
+        
+        parser.add_argument(
+            "--multiroot-data-file",
+            help="Path to an Xcode workspace to create a unified build of "
+            "SwiftSyntax with other projects.",
+        )
+        
+        parser.add_argument(
+            "--toolchain",
+            required=True,
+            help="The path to the toolchain that shall be used to build " "SwiftSyntax.",
+        )
+
     parser = argparse.ArgumentParser(
         formatter_class=argparse.RawDescriptionHelpFormatter, description=_DESCRIPTION
     )
 
     # -------------------------------------------------------------------------
+    # Shared arguments
 
     parser.add_argument(
         "-v", "--verbose", action="store_true", help="Enable verbose logging."
     )
 
+    sub_parsers = parser.add_subparsers()
+    
     # -------------------------------------------------------------------------
-    xcode_project_group = parser.add_argument_group("Xcode Project")
-
-    xcode_project_group.add_argument(
-        "--generate-xcodeproj",
-        action="store_true",
-        help="Generate an Xcode project for SwiftSyntax.",
-    )
+    generate_xcodeproj_parser = sub_parsers.add_parser("generate-xcodeproj", help="Generate an Xcode project for SwiftSyntax.")
+    generate_xcodeproj_parser.set_defaults(func=generate_xcodeproj)  
 
-    xcode_project_group.add_argument(
+    generate_xcodeproj_parser.add_argument(
         "--xcconfig-path",
         help="The path to an xcconfig file for generating Xcode projct.",
     )
 
     # -------------------------------------------------------------------------
-    build_group = parser.add_argument_group("Build")
-
-    build_group.add_argument(
-        "-r", "--release", action="store_true", help="Build in release mode."
-    )
-
-    build_group.add_argument(
-        "--build-dir",
-        default=None,
-        help="The directory in which build products shall be put. If omitted "
-        'a directory named ".build" will be put in the swift-syntax '
-        "directory.",
-    )
+    generate_source_code_parser = sub_parsers.add_parser("generate-source-code")
+    generate_source_code_parser.set_defaults(func=generate_source_code_command)
+    
+    default_build_arguments(generate_source_code_parser)
 
-    build_group.add_argument(
-        "--add-source-locations",
-        action="store_true",
-        help="Insert ###sourceLocation comments in generated code for "
-        "line-directive.",
+    generate_source_code_parser.add_argument(
+        "--gyb-exec",
+        default=GYB_EXEC,
+        help="Path to the gyb tool (default: %(default)s).",
     )
 
-    build_group.add_argument(
+    generate_source_code_parser.add_argument(
         "--degyb-only",
```

Comment:
Added `--degyb-only` to `generate-source-code` because in #381 you mention that it should not be implicit.
https://github.com/apple/swift-syntax/pull/381#discussion_r881655025